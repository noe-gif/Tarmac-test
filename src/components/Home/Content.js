/*
					üìú
		   
	************************************
	*                                  *
	*     DEFINES THE CONTENT FOR      *
	*      EACH NAV SECTIONS           *
	*                                  *
	************************************

*/

/* ‚öõÔ∏è react libraries */
import React, { useState, useEffect } from "react";

/* üé® assets libraries*/
import { ChevronLeftIcon, ChevronRightIcon } from '@heroicons/react/solid'
import ReactLoading from "react-loading";

/* üì¶ my components*/
import Flights from '../Arrivals/Arrivals';
import * as styler from './style';
import Filter from './Filter';


/* ____________________________________________________________________________________________________________________*/



/* ‚ùì props getting the nav section id so we know which one to display */
const Content = (props) => {
	const [currentPage, setCurrentPage] = useState(1);
	const [postsPerPage, setPostsPerPage] = useState(10);
	const [filter, setFilter] = useState('')
	let pageNumbers = [];
	const posts = props.flightsData === undefined ? [] : props.flightsData;
	const indexOfLastPost = currentPage * postsPerPage;
	const indexOfFirstPost = indexOfLastPost - postsPerPage;
	let currentPosts;

	const getFilter = (filterChosen) => {
		setFilter(filterChosen)
	}

	/* ‚ùì L56 - 71 */
	/* ---------------------------------------------------------------------------------------------------- */
	/* ---------------------  if (data loaded and no filters) {                             ----------------*/
	/* ---------------------      split the flights in 10 flights blocks                    ----------------*/
	/* ---------------------  } else if (there is a filter) {                               ----------------*/
	/* ---------------------      split the flights in n fights blocks depending on filter  ----------------*/
	/* ---------------------  } else {                                                      ----------------*/
	/* ---------------------      data did not load yet                                     ----------------*/
	/* ---------------------   }                                                            ----------------*/
	/* ---------------------------------------------------------------------------------------------------- */

	if (posts !== [] && posts.flights && posts.flights.data && filter === '') {
		pageNumbers = [];
		for (let i = 1; i <= Math.ceil(posts.flights.data.length / postsPerPage); i++) {
			pageNumbers.push(i);
		}
		currentPosts = posts.flights.data.slice(indexOfFirstPost, indexOfLastPost)
	} else if (filter !== '') {
		pageNumbers = [];
		let xx = posts.flights.data.filter((flight) => { if (flight.airline.name == filter) { return flight } })
		for (let i = 1; i <= Math.ceil(xx.length / postsPerPage); i++) {
			pageNumbers.push(i);
		}
		currentPosts = xx.slice(indexOfFirstPost, indexOfLastPost)
	} else {
		currentPosts = [];
	}


	/* ‚ùì each offset defines the content of each section on the left side*/
	const navSelectionContent = [
		props.flightsData.flights === undefined ? <ReactLoading type="spinningBubbles" color="#0000FF" style={styler.loadingStyle} />
			: <Flights posts={currentPosts} ></Flights>,
		<div style={{ width: '20%', height: '20%', position: 'absolute', top: '27%', left: '33%' }}><img alt="" src="https://user-images.githubusercontent.com/10260230/93533501-53aa0d80-f943-11ea-90d1-e6e70eca2e29.gif"></img></div>,
		<div style={{ width: '20%', height: '20%', position: 'absolute', top: '27%', left: '33%' }}><img alt="" src="https://i.imgur.com/VKfapZT.gif"></img></div>,
		<div style={{ width: '20%', height: '20%', position: 'absolute', top: '27%', left: '33%' }}><img alt="" src="https://stoic-liskov-800c94.netlify.app/images/sunset.gif"></img></div>,
		<div style={{ width: '20%', height: '20%', position: 'absolute', top: '27%', left: '33%' }}><img alt="" src="https://i.imgur.com/QW2GPll.gif"></img></div>,
		<div style={{ width: '20%', height: '20%', position: 'absolute', top: '28%', left: '33%' }}><img alt="" src="https://miro.medium.com/max/1400/1*KOAdvGvsKsFfWpmYTBpQFA.gif"></img></div>
	]
	return (
		<div className={styler.cssClass2}>

			{/* ‚ö†Ô∏è MAIN JSX CONTENT (WITHIN DASHED BORDERS) */}

			<main className={styler.cssClass3} style={styler.contentStyleN1} data-testid={`content-${1}`}>
				<div className={styler.cssClass5}>
					<div className={styler.cssClass6} >
						<div className={styler.cssClass7} style={styler.contentStyleN2} >
							{navSelectionContent[props.currentNavSelection]}
						</div>
					</div>
				</div>
			</main>
			{/* ‚ö†Ô∏è PAGINATION JSX */}

			{props.currentSection === 0 ? <>
				<div style={styler.paginationStyleN1} data-testid={`content-${2}`}>
					<div>
						<div style={{ marginLeft: '90px' }}>
							<p className={styler.cssClass8} style={posts !== [] && posts.flights && posts.flights.data ? {} : { marginLeft: '-180px' }}>
								Showing <span className={styler.cssClass9}>1</span> to <span className={styler.cssClass9}>10</span> of{' '}
								<span className={styler.cssClass9}>{posts !== [] && posts.flights && posts.flights.data ? posts.flights.data.length : '‚åõ'}</span> results
							</p>
						</div>
						<div style={{ height: '10px' }}></div>
						<div>
							<nav className={styler.cssClass10} style={{ marginLeft: '-50px' }} aria-label="Pagination">
								<a href="#" className={styler.cssClass1} onClick={() => { setCurrentPage(currentPage === 1 ? currentPage : currentPage - 1) }}>
									<span className={styler.cssClass11}>Previous</span>
									<ChevronLeftIcon className={styler.cssClass12} aria-hidden="true" />
								</a>
								{pageNumbers.map(number =>
									<a onClick={() => { setCurrentPage(number) }} key={number} href="!#" className={number === currentPage ? styler.classSelected : styler.classNotSelected}>{number}</a>
								)}
								<a href="#" className={styler.cssClass4} onClick={() => { setCurrentPage(currentPage === 10 ? currentPage : currentPage + 1) }}>
									<span className={styler.cssClass11}>Next</span>
									<ChevronRightIcon className={styler.cssClass12} aria-hidden="true" />
								</a>
							</nav>
						</div>
					</div>
				</div>
				<Filter getFilter={getFilter} airlinesNames={props.companiesData} flightsData={props.flightsData} /></> : <></>}
		</div>
	)
}

export default Content;