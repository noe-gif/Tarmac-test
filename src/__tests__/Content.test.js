import { render, screen, cleanup } from '@testing-library/react';
import { useDispatch, useSelector, Provider } from "react-redux";
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer' // snapshot
import Content from '../components/Home/Content'


test('--------- CONTENT TEST #1 => PAGINATION ---------', () => {
	render(<Content flightsData={{}} companiesData={{}} currentNavSelection={[]} currentSection={0}></Content>);
	const contentElement = screen.getByTestId(`content-${1}`)
	const paginationElement = screen.getByTestId(`content-${2}`)
	expect(contentElement).toBeInTheDocument();
	expect(paginationElement).toBeInTheDocument();
	expect(paginationElement).toHaveTextContent('Showing');
	expect(paginationElement).toHaveTextContent('1');
	expect(paginationElement).toHaveTextContent('to');
	expect(paginationElement).toHaveTextContent('10');
	expect(paginationElement).toHaveTextContent('of');
	expect(paginationElement).toHaveTextContent('results');
})

test('match snapshot', () => {
	const tree = renderer.create(<Content flightsData={{}} companiesData={{}} currentNavSelection={[]} currentSection={0}></Content>).toJSON();
	expect(tree).toMatchSnapshot();
})