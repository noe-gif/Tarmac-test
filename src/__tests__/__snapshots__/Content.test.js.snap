// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="flex flex-col w-0 flex-1 overflow-hidden"
>
  <main
    className="flex-1 relative z-0 overflow-y-auto focus:outline-none"
    data-testid="content-1"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="max-w-7xl mx-auto px-4 sm:px-6 md:px-8"
    >
      <div
        className="py-4"
      >
        <div
          className="border-4 border-dashed border-gray-200 rounded-lg h-96"
          style={
            Object {
              "height": "780px",
              "marginLeft": "-250px",
              "marginTop": "50px",
              "width": "1200px",
            }
          }
        />
      </div>
    </div>
  </main>
  <div
    data-testid="content-2"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100px",
        "justifyContent": "center",
        "left": "40%",
        "position": "fixed",
        "top": "88%",
        "width": "500px",
      }
    }
  >
    <div>
      <div
        style={
          Object {
            "marginLeft": "90px",
          }
        }
      >
        <p
          className="text-sm text-gray-700"
          style={
            Object {
              "marginLeft": "-180px",
            }
          }
        >
          Showing 
          <span
            className="font-medium"
          >
            1
          </span>
           to 
          <span
            className="font-medium"
          >
            10
          </span>
           of
           
          <span
            className="font-medium"
          >
            âŒ›
          </span>
           results
        </p>
      </div>
      <div
        style={
          Object {
            "height": "10px",
          }
        }
      />
      <div>
        <nav
          aria-label="Pagination"
          className="relative z-0 inline-flex rounded-md shadow-sm -space-x-px"
          style={
            Object {
              "marginLeft": "-50px",
            }
          }
        >
          <a
            className="relative inline-flex items-center px-2 py-2 rounded-l-md border border-gray-300 bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
            href="#"
            onClick={[Function]}
          >
            <span
              className="sr-only"
            >
              Previous
            </span>
            <svg
              aria-hidden="true"
              className="h-5 w-5"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
                fillRule="evenodd"
              />
            </svg>
          </a>
          <a
            className="relative inline-flex items-center px-2 py-2 rounded-r-md border border-gray-300 bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
            href="#"
            onClick={[Function]}
          >
            <span
              className="sr-only"
            >
              Next
            </span>
            <svg
              aria-hidden="true"
              className="h-5 w-5"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
                fillRule="evenodd"
              />
            </svg>
          </a>
        </nav>
      </div>
    </div>
  </div>
  <div
    className="relative inline-block text-left"
    data-testid="filter-1"
    style={
      Object {
        "left": "82%",
        "position": "absolute",
        "top": "7%",
      }
    }
  >
    <div>
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="inline-flex justify-center w-full rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-sm font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-100 focus:ring-indigo-500"
        id="headlessui-menu-button-2"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        Options
        <svg
          aria-hidden="true"
          className="-mr-1 ml-2 h-5 w-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
